operators/src/util/raster_stream_to_geotiff.rs

```diff
@@ -0,0 +1,231 @@
+        gdal_writer(
+            &rx,
+            &file_name_clone,
+            query_rect,
+            width,
+            height,
+            no_data_value,
+            spatial_reference,
+        )
+    });
+
+    let mut tile_stream = processor.raster_query(query_rect, &query_ctx).await?;
+
+    while let Some(tile) = tile_stream.next().await {
+        tx.send(tile?).map_err(|_| Error::ChannelSend)?;
+    }
+
+    drop(tx);
+
+    writer.await??;
+
+    // TODO: use higher level rust-gdal method when it is mapped
+    let bytes = get_vsi_mem_file_bytes_and_free(&file_name);
+
+    Ok(bytes)
+}
+
+fn gdal_writer<T: Pixel + GdalType>(
+    rx: &Receiver<RasterTile2D<T>>,
+    file_name: &str,
+    query_rect: QueryRectangle,
+    width: u32,
+    height: u32,
+    no_data_value: Option<f64>,
+    spatial_reference: SpatialReference,
+) -> Result<()> {
+    let x_pixel_size = query_rect.bbox.size_x() / f64::from(width);
+    let y_pixel_size = query_rect.bbox.size_y() / f64::from(height);
+
+    let output_geo_transform =
+        GeoTransform::new(query_rect.bbox.upper_left(), x_pixel_size, -y_pixel_size);
+    let output_bounds = query_rect.bbox;
+
+    let driver = Driver::get("GTiff")?;
+    // TODO: "COMPRESS, DEFLATE" flags but rust-gdal doesn't support setting this yet(?)
+    let mut dataset =
+        driver.create_with_band_type::<T>(&file_name, width as isize, height as isize, 1)?;
+
+    dataset.set_spatial_ref(&spatial_reference.try_into()?)?;
+    dataset.set_geo_transform(&output_geo_transform.into())?;
+    let mut band = dataset.rasterband(1)?;
+
+    if let Some(no_data) = no_data_value {
+        band.set_no_data_value(no_data)?;
+    }
+
+    while let Ok(tile) = rx.recv() {
+        let tile_info = tile.tile_information();
+
+        let tile_bounds = tile_info.spatial_bounds();
+
+        let mat_tile = if output_bounds.contains_bbox(&tile_bounds) {
+            tile.into_materialized_tile()
+        } else {
+            // extract relevant data from tile (intersection with output_bounds)
+
+            // TODO: snap intersection to pixels?
+            let intersection = output_bounds
+                .intersection(&tile_bounds)
+                .expect("tile must intersect with query");
+
+            let shape: GridShape2D = [
+                (intersection.size_y() / y_pixel_size).ceil() as usize,
+                (intersection.size_x() / x_pixel_size).ceil() as usize,
+            ]
+            .into();
+
+            let output_grid = Grid2D::new_filled(
+                shape,
+                no_data_value.map_or_else(T::zero, T::from_),
+                no_data_value.map(T::from_),
+            );
+
+            let output_geo_transform = GeoTransform {
+                origin_coordinate: intersection.upper_left(),
+                x_pixel_size,
+                y_pixel_size: -y_pixel_size,
+            };
+
+            let mut output_tile = RasterTile2D::new_without_offset(
```

Comment:
you can skip this and use code from blit.rs line 35. yust create a grid with offset and blit the data into that.

Reply:
I wanted to re-use that functionality. I need a Tile anyway for computing window_size and window further down and this requires computing the geo_transform anyway

Reply:
i think you can do it without the tile since the window is the overlapping area in pixels

Reply:
cf my other comment

Reply:
8266cd5e3c70066c759d83db427caf5b3a25ac88